### PR TITLE
Add support for BadRequest in RazorPages

### PR DESCRIPTION
**Pull request by jbagga**
Addresses #7095

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.RazorPages/PageBase.cs [hunk 1]
```diff
@@ -145,6 +145,39 @@ public override void EndContext()
...
+        /// <param name="error">An error object to be returned to the client.</param>
+        /// <returns>The created <see cref="BadRequestObjectResult"/> for the response.</returns>
+        [NonAction]
+        public virtual BadRequestObjectResult BadRequest(object error)
+            => new BadRequestObjectResult(error);
+
+        /// <summary>
+        /// Creates a <see cref="BadRequestObjectResult"/> that produces a <see cref="StatusCodes.Status400BadRequest"/> response.
+        /// </summary>
+        /// <param name="modelState">The model state dictionary containing errors to be returned to the client.</param>
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/PageBase.cs]: > The `<see cref="ModelStateDictionary" />` ...

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs [hunk 2]
```diff
@@ -291,6 +291,20 @@ public RazorPagesTest(MvcTestFixture<RazorPagesWebSite.Startup> fixture)
             Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
         }
 
+        [Fact]
+        public async Task NullParameter_BadRequest()
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs]: `PageHandlerCanReturnBadRequest` (the current name doesn't say what's being tested).

File: test/WebSites/RazorPagesWebSite/Pages/HandlerWithParameter.cshtml.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        {
+            if (testParameter == null)
+            {
+                return BadRequest("Parameter cannot be null.");
+            }
+
+            return Page();
+        }
+    }
+}
```
Comment (pranavkm) [inline test/WebSites/RazorPagesWebSite/Pages/HandlerWithParameter.cshtml.cs]: new line at EOF

File: src/Microsoft.AspNetCore.Mvc.RazorPages/PageBase.cs (hunk 1 above)
Comment (jbagga) [reply src/Microsoft.AspNetCore.Mvc.RazorPages/PageBase.cs]: 👍 Changing it in [here](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs#L1753) in `ControllerBase` as well

**Pull request merged**